.github/workflows/cicd-main.yml

```diff
@@ -22,12 +22,46 @@ on:
   push:
     branches:
       - 'main'
+  workflow_dispatch:
+    inputs:
+      test_to_run:
+        required: false
+        default: all
+        type: string
+        description: Comma-separated list of tests to run. Use "all" to run the full test suite.
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
-jobs:
+jobs: 
+  pre-flight:
+    runs-on: ubuntu-latest
+    outputs:
+      test_to_run: ${{ steps.test_to_run.outputs.main }}
+      all: ${{ steps.all.outputs.main }}
+    steps:
+      - name: Parse test_to_run
+        id: test_to_run
+        run: |
+          parsed_string=$(echo ${{ inputs.test_to_run || 'all' }} | jq -c --raw-input 'split(",")')
+          echo "main=${parsed_string}" | tee -a "$GITHUB_OUTPUT"
```

Comment:
nit: maybe renaming this to `tests_as_array` since it semantically looks like it overlaps with the `main` in the step below

Reply:
I'd prefer to do this separately since I'm consolidating/refactoring all workflows currently anyway. Want to avoid to many heterogenous workflows at this point to ease the consolidation into the workflows repo